cmd/context/create.go

```diff
@@ -196,6 +194,34 @@ func (c *ContextCommand) UseContext(ctx context.Context, ctxOptions *ContextOpti
 	return nil
 }
 
+func hasAccessToNamespace(ctx context.Context, c *ContextCommand, ctxOptions *ContextOptions) (bool, error) {
```

Comment:
I think we should split this function into 2:
1. creation of `access` function
2. running the `access` function

In this way we would not need to modify `utils.HasAccessToK8sClusterNamespace` to be a variable, we can leave it as a function and then inject the `access` function we'd like

We could also separate the testing of `creation` and `execution` to have smaller tests